sources/org.osbuild.ostree

```diff
@@ -97,29 +92,33 @@ def download(items, cache):
         ostree("remote", "add",
                uid, url,
                *verify_args,
-               repo=repo_cache)
+               repo=self.repo_cache)
 
         for key in gpg:
             ostree("remote", "gpg-import", "--stdin", uid,
-                   repo=repo_cache, _input=key)
+                   repo=self.repo_cache, _input=key)
 
         # Transfer the commit: remote → cache
         print(f"pulling {commit}", file=sys.stderr)
-        ostree("pull", uid, commit, repo=repo_cache)
+        ostree("pull", uid, commit, repo=self.repo_cache)
 
         # Remove the temporary remotes again
         ostree("remote", "delete", uid,
-               repo=repo_cache)
-
-
-class OSTreeSource(sources.SourceService):
+               repo=self.repo_cache)
 
-    def download(self, items, cache, _options):
-        cache = os.path.join(cache, "org.osbuild.ostree")
-        os.makedirs(cache, exist_ok=True)
+    def config_download(self):
+        # Prepare the cache and the output repo
+        self.repo_cache = os.path.join(self.cache, "repo")
+        ostree("init", mode="archive", repo=self.repo_cache)
 
-        download(items, cache)
+        # Make sure the cache repository uses locks to protect the metadata during
+        # shared access. This is the default since `2018.5`, but lets document this
+        # explicitly here.
+        ostree("config", "set", "repo.locking", "true", repo=self.repo_cache)
 
+    # pylint: disable=[no-self-use]
+    def exists(self, _checksum, _item):
+        return False
```

Comment:
I think we can use `osbuild.util.rev_parse` to check if an object exists in the cache. We could add a `check = True` argument to it, so we don't have to throw and catch an exception but just return the return-value.